helpers/calendar/ratios_and_utilization.py

```diff
@@ -60,53 +66,68 @@ def get_analytics_ratios_per_room(self, query, start, end):
             events_list = []
             checkins = 0
             cancellations = 0
+            calendar_events_list = []
+            app_events_list = []
 
-            checkins, cancellations, events_list = RoomAnalyticsRatios.retrieve_cancellations_and_checkins_for_room(self,  # noqa
+            checkins, cancellations, events_list, calendar_events_list = RoomAnalyticsRatios.retrieve_cancellations_and_checkins_for_room(self,  # noqa
                                                         room['calendar_id'],
                                                         start_date,
                                                         day_after_end_date,
                                                         checkins,
                                                         cancellations,
-                                                        events_list)
+                                                        events_list,
+                                                        calendar_events_list,
+                                                        app_events_list)
 
             ratio_object = RoomAnalyticsRatios.map_results_to_ratio_class(self,  # noqa
                                 checkins,
                                 cancellations,
                                 events_list,
+                                calendar_events_list,
+                                app_events_list,
                                 room['name'])
 
             response.append(ratio_object)
         return response
 
-    def retrieve_cancellations_and_checkins_for_room(self, calendar_id, start_date, day_after_end_date, checkins,cancellations, events_list):  # noqa
+    def retrieve_cancellations_and_checkins_for_room(self, calendar_id, start_date, day_after_end_date, checkins, cancellations, events_list, calendar_events_list, app_events_list):  # noqa
         """ Retrieve cancellations and checkins for a room
         :params
             - calendar_id, start_date, day_after_end_date,
                 checkins,cancellations, output
         """
-        calendar_events = CommonAnalytics.get_all_events_in_a_room(
+        all_events = CommonAnalytics.get_all_events_in_a_room(
                 self, calendar_id, start_date, day_after_end_date)
-        if calendar_events:
-            for event in calendar_events:
-                if event.get('attendees'):
-                    event_details = CommonAnalytics.get_event_details(
+        if all_events:
```

Comment:
This block has too many if statements i.e it is deeply nested. kindly consider refactoring it to avoid this.

Reply:
This has been refactored.